core/web/router.go

```diff
@@ -232,6 +232,10 @@ func v2Routes(app chainlink.Application, r *gin.RouterGroup) {
 	))
 	{
 		uc := UserController{app}
+		authv2.GET("/users", auth.RequiresAdminRole(uc.Index))
+		authv2.POST("/users", auth.RequiresAdminRole(uc.Create))
+		authv2.PATCH("/users", auth.RequiresAdminRole(uc.Update))
+		authv2.DELETE("/users/:email", auth.RequiresAdminRole(uc.Delete))
```

Comment:
Did you consider using the middleware api a little more directly? e.g.
```suggestion
		users := authv2.Group("/users", auth.RequiresAdminRole)
		users..GET("", uc.Index)
		users.POST("", uc.Create)
		users.PATCH("", uc.Update)
		users.DELETE("/:email", uc.Delete)
```
Or for individual ones:
```go
		authv2.Use(auth.RequiresAdminRole).PATCH("/users", uc.Update)
```
I'm not sure exactly what the tradeoffs are other than convenience/readability, but it seems like we should follow the composable router API if we can :shrug:

Reply:
I think thats a good idea to use the standard routing convention this way, I would prefer the second suggestion where we have the granularity for individual routes instead of grouping the routes by permission, the current grouping makes sense per feature, so I think it is more straightforward to extend those lines to use role assertion middleware directly. 

I tried to update all to `authv2.Use(auth.RequiresAdminRole).{{route}}` pattern, but gin does not like the setup we have with the `for` loop setting route handlers, the current set up we have re-registered the new middleware I defined. I don't have a great solution for that besides unrolling the for loop and having a flat verbose list instead. I reverted the change, the current implementation feels flexible and not obtuse in my opinion 🙂